trin-history/src/jsonrpc.rs

```diff
@@ -193,13 +192,20 @@ impl HistoryRequestHandler {
                     None
                 }
             };
-        // Get data from peer
-        let (possible_content_bytes, closest_nodes) = match local_content {
-            Some(val) => (Some(val), vec![]),
+        let (possible_content_bytes, trace) = match local_content {
+            Some(val) => {
+                let local_enr = self.network.overlay.local_enr();
+                let mut trace = QueryTrace::new(
+                    &self.network.overlay.local_enr(),
+                    NodeId::new(&content_key.content_id()).into(),
+                );
+                trace.node_responded_with_content(&local_enr);
+                (Some(val), if is_trace { Some(trace) } else { None })
```

Comment:
Seems like a bummer to do all the trace work every time, even when it wasn't requested. Maybe something like:
```suggestion
                let trace_option = if is_trace {
                    let mut trace = QueryTrace::new(
                        &self.network.overlay.local_enr(),
                        NodeId::new(&content_key.content_id()).into(),
                    );
                    trace.node_responded_with_content(&local_enr);
                    Some(trace)
                } else { None }
                (Some(val), trace_option)
```